MoneyDataType/Amount.cs

```diff
@@ -1,141 +1,145 @@
+using Money.Abstractions;
+using Money.Serialization;
 using System;
 using System.Diagnostics;
 using System.Globalization;
 using System.Text.Json.Serialization;
-using Money.Abstractions;
-using Money.Serialization;
 
-namespace Money
+namespace Money;
+
+/// <summary>
+/// A money amount, which is represented by a decimal number and a currency.
+/// </summary>
+[JsonConverter(typeof(AmountConverter))]
+[Newtonsoft.Json.JsonConverter(typeof(LegacyAmountConverter))]
+[DebuggerDisplay("{DebuggerDisplay,nq}")]
+public readonly struct Amount : IEquatable<Amount>, IComparable<Amount>
 {
-    /// <summary>
-    /// A money amount, which is represented by a decimal number and a currency
-    /// </summary>
-    [JsonConverter(typeof(AmountConverter))]
-    [Newtonsoft.Json.JsonConverter(typeof(LegacyAmountConverter))]
-    [DebuggerDisplay("{DebuggerDisplay,nq}")]
-    public struct Amount : IEquatable<Amount>, IComparable<Amount>
+    public Amount()
+        : this(0, Money.Currency.UnspecifiedCurrency)
     {
-        public Amount() {
-            Value = 0;
-            Currency = Money.Currency.UnspecifiedCurrency;
-        }
+    }
 
-        public Amount(decimal value, RegionInfo region)
-        {
-            if (region == null)
-                throw new ArgumentNullException(nameof(region));
+    public Amount(decimal value, RegionInfo region)
+        : this(value, Money.Currency.FromRegion(region))
+    {
+    }
 
-            Currency = Money.Currency.FromRegion(region);
-            Value = value;
-        }
+    public Amount(decimal value, CultureInfo culture)
+        : this(value, Money.Currency.FromCulture(culture))
+    {
+    }
 
-        public Amount(decimal value, CultureInfo culture)
-        {
-            if (culture is null)
-                throw new ArgumentNullException(nameof(culture));
+    public Amount(decimal value, ICurrency currency)
+    {
+        ArgumentNullException.ThrowIfNull(currency);
 
-            Currency = Money.Currency.FromCulture(culture);
-            Value = value;
-        }
+        Currency = currency;
+        Value = value;
+    }
 
-        /// <summary>
-        /// Constructs a new money amount
-        /// </summary>
-        /// <param name="value">The decimal value</param>
-        /// <param name="currency">The currency</param>
-        public Amount(decimal value, ICurrency currency)
-        {
-            Currency = currency ?? throw new ArgumentNullException(nameof(currency));
-            Value = value;
-        }
+    /// <summary>
+    /// Gets the decimal value.
+    /// </summary>
+    public decimal Value { get; }
```

Comment:
When you say "R#" I take it you refer to ReSharper (specifically [this](https://www.jetbrains.com/help/resharper/File_and_Type_Layout.html)), right? Took me a bit to ponder what the [.Net implementation of the R programming language](https://github.com/rsharp-lang/R-sharp) has to do with file layouts. :D

Can you send over the file layout you've created? I'm curious (and also wanna see if I can make it work in Rider).

Reply:
Yeah, that makes sense. It was just unfamiliar.

Thanks for the config, I've fed it into Rider's code cleanup tool and it did some cool magic! Fixed [a bunch of files](https://github.com/OrchardCMS/OrchardCore.Commerce/commit/776223fb12ac4639d0e3879c12f66649a6e966be). I have tested it on the _Amount.cs_ from dev too and it reordered seemingly fine for me, so I don't know why it didn't work for you.
I don't trust it fully, so I've reviewed everything manually, but only had to add/remove blank lines around the moved members for esthetics. And because it wants to collapse the braces of empty classes into a one-liner `{}` for some reason. A small price to pay for hours of tedium avoided. Thanks a lot, I will use this!

Have you uploaded this layout file to the wiki? If not, would you mind if I did and referenced it in the [Rider page](https://lombiq.atlassian.net/wiki/spaces/DEV/pages/2495578139/Rider)? It should be super useful.

Reply:
I've been using this file layout configuration for over a year now, and can't complain! Except for that it wouldn't run in OCC for some reason. But that's a different story.

Feel free to post the config wherever you like 😀 I'd be more than glad if it proved to be of use for others.

Concerning `{}`, have you configured Rider correctly to put them on separate lines, independently of the file layout?